### ISSUE TITLE
Grabbing input_datetime helper that has both a date and time through GetDateTimeEntity fails to parse

### ISSUE THREAD

**Issue by Onaka** [leosperry/ha-kafka-net#77]
Given input_datetime.last_snowshovel is a datetime helper with both the time and the date selected:
```csharp
var snowshovel = await _services.EntityProvider.GetDateTimeEntity("input_datetime.last_snowshovel");
```
The above fails with a parsing error.

Error for doing it this way is as follows:
```json
{
  "type": "System.Text.Json.JsonException",
  "message": "The JSON value could not be converted to System.Nullable`1[System.DateTime]. Path: $.state | LineNumber: 0 | BytePositionInLine: 75.",
  "stackTrace": "   at System.Text.Json.ThrowHelper.ReThrowWithPath(ReadStack& state, Utf8JsonReader& reader, Exception ex)\r\n   at System.Text.Json.Serialization.JsonConverter`1.ReadCore(Utf8JsonReader& reader, JsonSerializerOptions options, ReadStack& state)\r\n   at System.Text.Json.Serialization.Metadata.JsonTypeInfo`1.ContinueDeserialize(ReadBufferState& bufferState, JsonReaderState& jsonReaderState, ReadStack& readStack)\r\n   at System.Text.Json.Serialization.Metadata.JsonTypeInfo`1.Deserialize(Stream utf8Json)\r\n   at HaKafkaNet.HaApiProvider.GetEntity[T](String entity_id, CancellationToken cancellationToken)",
  "innerException": {
    "type": "System.FormatException",
    "message": "The JSON value is not in a supported DateTime format.",
    "stackTrace": "   at System.Text.Json.ThrowHelper.ThrowFormatException(DataType dataType)\r\n   at System.Text.Json.Serialization.Converters.DateTimeConverter.Read(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options)\r\n   at System.Text.Json.Serialization.Converters.NullableConverter`1.Read(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options)\r\n   at System.Text.Json.Serialization.Metadata.JsonPropertyInfo`1.ReadJsonAndSetMember(Object obj, ReadStack& state, Utf8JsonReader& reader)\r\n   at System.Text.Json.Serialization.Converters.ObjectDefaultConverter`1.OnTryRead(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options, ReadStack& state, T& value)\r\n   at System.Text.Json.Serialization.JsonConverter`1.TryRead(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options, ReadStack& state, T& value, Boolean& isPopulatedValue)\r\n   at System.Text.Json.Serialization.JsonConverter`1.ReadCore(Utf8JsonReader& reader, JsonSerializerOptions options, ReadStack& state)",
    "innerException": null,
    "innerExceptions": null
  },
  "innerExceptions": null
}
```

State of input_datetime.last_snowshovel at time of error:
`2024-10-09 22:00:00`
Attributes of input_datetime.last_snowshovel at time of error:
```yaml
has_date: true
has_time: true
editable: true
year: 2024
month: 10
day: 9
hour: 22
minute: 0
second: 0
timestamp: 1728500400
icon: mdi:shovel
friendly_name: Last snowshovel
```

Workaround I'm using for now:
```csharp

public static DateTime UnixTimeStampToDateTime( double unixTimeStamp )
{
    DateTime dateTime = new DateTime(1970, 1, 1, 0, 0, 0, 0, DateTimeKind.Utc);
    dateTime = dateTime.AddSeconds( unixTimeStamp ).ToLocalTime();
    return dateTime;
}

var snowshovel = await _services.EntityProvider.GetEntity("input_datetime.last_snowshovel");
DateTime snowshovelDateTime = UnixTimeStampToDateTime(snowshovel.Attributes.GetProperty("timestamp").GetDouble());
```

**Comment by leosperry** [comment]
Excellent find. I'll get a fix out this weekend. Thank you for reporting it.
Also feel free to join the Discord linked on the main readme. I'd love to get more feedback on features people are using.

**Comment by leosperry** [comment]
The reason for this bug is that the Microsoft built `JsonSerializer` only supports ISO 8601 date time formats and Home Assistant uses a simpler standard MM/DD/YYYY HH:MM format.

Resolution: 
Create a `JsonConverter<DateTime?>` that uses standard `DateTime.Parse()` to support more formats

**Comment by leosperry** [comment]
Fixed in Version 9.2
